system/lib/wasmfs/wasmfs.cpp

```diff
@@ -62,9 +62,55 @@ extern "C" void wasmfs_flush(void) {
   // Flush musl libc streams.
   fflush(0);
 
-  // Flush our own streams. TODO: flush all backends.
+  // Flush our own streams.
   (void)SpecialFiles::getStdout()->locked().flush();
   (void)SpecialFiles::getStderr()->locked().flush();
+
+  std::vector<std::shared_ptr<Directory>> toFlush;
+  toFlush.push_back(wasmFS.getRootDirectory());
+
+  while (!toFlush.empty()) {
+    auto dir = toFlush.back();
+    toFlush.pop_back();
+
+    auto lockedDir = dir->locked();
+    Directory::MaybeEntries entries = lockedDir.getEntries();
+    if (int err = entries.getError()) {
+#ifndef NDEBUG
+      std::string errorMessage =
+        "Non-fatal error code " + std::to_string(err) +
+        " while flushing directory: " + lockedDir.getName(dir);
+      emscripten_console_error(errorMessage.c_str());
+#endif
+      continue;
+    }
+    // TODO: Investigate why *lockedDir.getEntries() does not go through the loop.
+    for (auto& entry : *entries) {
+      auto child = lockedDir.getChild(entry.name);
+      if (!child) {
+#ifndef NDEBUG
+          std::string errorMessage = "Child: " +
+                                     entry.name +
+                                     " was null while flushing directory: " + lockedDir.getName(dir);
+          emscripten_console_error(errorMessage.c_str());
+#endif
```

Comment:
```suggestion
        // This file must have been deleted between the call to `getEntries()` and now.
```

I don't think we need to treat this as an error condition, even with assertions enabled. A comment explaining why this situation may arise should be sufficient.